src/Symfony/Component/Config/ConfigCacheFactory.php

```diff
@@ -35,12 +35,8 @@ public function __construct(bool $debug)
     /**
      * {@inheritdoc}
      */
-    public function cache(string $file, $callback)
+    public function cache(string $file, callable $callback)
     {
-        if (!\is_callable($callback)) {
-            throw new \InvalidArgumentException(sprintf('Invalid type for callback argument. Expected callable, but got "%s".', \gettype($callback)));
-        }
```

Comment:
With the `callable` type-hint we will get a `TypeError` now. So no need to check the argument anymore.